packages/components/src/radio-group/radio.tsx

```diff
@@ -0,0 +1,45 @@
+/**
+ * WordPress dependencies
+ */
+import { forwardRef, useContext } from '@wordpress/element';
+
+/**
+ * External dependencies
+ */
+// eslint-disable-next-line no-restricted-imports
+import * as Ariakit from '@ariakit/react';
+
+/**
+ * Internal dependencies
+ */
+import Button from '../button';
+import { RadioGroupContext } from './context';
+import type { WordPressComponentProps } from '../context';
+import type { RadioProps } from './types';
+
+function UnforwardedRadio(
+	{ value, children }: WordPressComponentProps< RadioProps, 'button', false >,
```

Comment:
the older version of the component was forwarding all remaining props to the underlying radio button — we should do the same here. Ie.

```suggestion
	{ value, children, ...props }: WordPressComponentProps< RadioProps, 'button', false >,
```

Reply:
So we tried that but as we're using a `button` in the `render` prop but `Ariakit.RadioGroup` expects to be an `input` element TypeScript complains about incompatible properties. If we change `button` to `input` on line 21 though TS doesn't complain anymore. However, as we're using a `Button` component inside the render prop we'd ship wrong types. @diegohaz do you, by any chance, have a recommendation on how to proceed?

Reply:
Pass the props to `Button`:

```jsx
<Ariakit.Radio
  render={<Button {...props} />}
>
```